change/@ni-nimble-components-bcea55bf-ed59-4b11-b80b-6311749da460.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```

Comment:
In macOS VoiceOver when the filter mode is `none` and I arrow between items, it reads the item names. But when I set filter mode to `standard`, it doesn't say anything when I arrow between items. I'm guessing it's because focus is still on the search input.

This seems like an important interaction for screenreader users; without I'm not sure how they'd tell what they've selected. (It's possible there are other gestures I'm not aware of) But I tried the same on the combobox and our behavior there is broken too (it reads the previously selected item for some reason).

Reply:
In my research into getting this to work as expected I found that it is common (if not universal) that the input element where a user enters text applies the `aria-controls` and `aria-activedescendant` attributes with the appropriate element IDs. This alone, however, was not enough to get any screen reader (I used NVDA and Windows Narrator) to provide appropriate voice over as well as manage the expected interactive state.

The extra step I could take that would make the screen readers work well in Chromium-based browsers (but not Firefox) was by setting `type="search"` on the `input` element. The drawback to this, however, was that an 'X' glyph was then displayed by the browser, that did not allow for a browser-agnostic way to remove.

The accessibility tree for what we are doing and what is done in screen reader-friendly selects seems consistent. The main difference that I can see as possibly messing things up is the fact that the elements being navigated between are in the shadow DOM.

Also note that the current FAST `Combobox` has similar issues working with screen readers, so this problem extends beyond our own implementation.

For now I think we should move forward with the current approach and then decide what the appropriate action to take is regarding this problem (more research, filing issues with screen readers, etc...).